api/types/app.go

```diff
@@ -301,14 +301,16 @@ func (a *AppV3) CheckAndSetDefaults() error {
 	return nil
 }
 
-// DeduplicateApps deduplicates apps by name.
+// DeduplicateApps deduplicates apps by combination of app name and public address.
+// Apps can have the same name but also could have different addresses.
 func DeduplicateApps(apps []Application) (result []Application) {
 	seen := make(map[string]struct{})
 	for _, app := range apps {
-		if _, ok := seen[app.GetName()]; ok {
+		key := fmt.Sprintf("%s%s", app.GetName(), app.GetPublicAddr())
```

Comment:
these are great edge case questions i never thought about, and i'm not sure it was ever accounted for. the scope for this requires how dedupping is currently working for the CLI as well (i think), and i spoke with @zmb3 on my 1x1 and he said this can be done in another issue where while we dedup we check `if dup, check for later expiry date`, is this okay?